src/ghvmctl

```diff
@@ -19,49 +26,65 @@ prepare_vm() {
 		-d root,size="${VM_DISK_GIB}GiB"
 
 	# Wait for the VM agent to be running
-	while ! lxc exec test-vm -- cat /etc/hostname &>/dev/null; do
-		sleep 2
+	while ! lxc exec "${VM_NAME}" -- true 2>/dev/null; do
+		sleep 1
 	done
 
 	info "Installing utilities in VM"
 
+	# Try avoid gnome-initial-setup to start.
+	lxc exec "${VM_NAME}" -- runuser -u ubuntu -- \
+		sh -c 'echo yes > ~/.config/gnome-initial-setup-done'
+	lxc exec "${VM_NAME}" -- sh -c \
+		'rm /usr/lib/systemd/user/gnome-session.target.wants/gnome-initial-setup*'
+	lxc exec "${VM_NAME}" -- sh -c \
+		'rm /etc/xdg/autostart/gnome-initial-setup*'
+
```

Comment:
Is this not inherently racy? Does this reliably prevent the thing from running?

Reply:
Ah I see you're stopping it further down - in that case do we need these lines if we're just going to explicitly stop the service anyway?

Reply:
So, as mentioned in the commit message... Sadly it is racy.

In my tests it 99% avoided it to run, so I feel it's still better to have, since it may still prevent things to run for being killed at later point, and it's a more reliable way of stopping compared to the one below (or the one we had).

However, my initial idea was to do something like:

```bash
        info "Creating VM '${VM_NAME}' with LXD"
        lxc init -q "images:ubuntu/${VM_SERIES}/desktop" "${VM_NAME}" --vm \
                -c limits.cpu="${VM_CPUS}" \
                -c limits.memory="${VM_MEM_GIB}GiB" \
                -d root,size="${VM_DISK_GIB}GiB"

        info "Configuring VM '${VM_NAME}'"
        config_folder=$(mktemp -d "$PWD"/config-XXXXXX)
        echo "yes" > "$config_folder/gnome-initial-setup-done"
        lxc config -q device add "${VM_NAME}" ubuntuconfig disk \
                source="$config_folder" path=/home/ubuntu/.config
 
        info "Launching VM '${VM_NAME}'"
        lxc start -q "${VM_NAME}"
```

However, this does not work properly since lxc user has no powers to read the mounted config folder, maybe there are other tricks like this we could use better maksing the systemd job instead.

I'm open to further suggestions, but after various rebuilds I had no more time and I accepted the more "reacy" solution.

Reply:
Okay, let's go with this for now and keep an eye on it, thanks! :)